File: src/main/java/net/insprill/cjm/hooks/AuthMeHook.java
package net.insprill.cjm.hooks;

import fr.xephi.authme.api.v3.AuthMeApi;
import fr.xephi.authme.events.LoginEvent;
import net.insprill.cjm.handlers.PlayerHandler;
import net.insprill.cjm.messages.MessageAction;
import net.insprill.cjm.messages.MessageSender;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class AuthMeHook implements Listener {

    @EventHandler(priority = EventPriority.MONITOR)
    public void onLogin(LoginEvent e) {
        sendMessage(e.getPlayer());
    }

    @EventHandler(priority = EventPriority.MONITOR)
    public void onJoin(PlayerJoinEvent e) {
        e.setJoinMessage("");
        if (AuthMeApi.getInstance().isUnrestricted(e.getPlayer())) {
            sendMessage(e.getPlayer());
        }
    }

    private void sendMessage(Player player) {
        PlayerHandler.setStatus(player.getUniqueId(), PlayerHandler.Status.LOGGED_IN);
        MessageSender.getInstance().sendMessages(player, (player.hasPlayedBefore()) ? MessageAction.JOIN : MessageAction.FIRST_JOIN, true);
    }

}

File: src/main/kotlin/net/insprill/cjm/compatibility/hook/AuthMeHook.kt
package net.insprill.cjm.compatibility.hook

import fr.xephi.authme.api.v3.AuthMeApi
import fr.xephi.authme.events.LoginEvent
import net.insprill.cjm.CustomJoinMessages
import net.insprill.cjm.messages.MessageAction
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class AuthMeHook(private val plugin: CustomJoinMessages) : Listener {

    @EventHandler(priority = EventPriority.MONITOR)
    fun onLogin(e: LoginEvent) {
        sendMessage(e.player)
    }

    @EventHandler(priority = EventPriority.MONITOR)
    fun onJoin(e: PlayerJoinEvent) {
        e.joinMessage = ""
        if (AuthMeApi.getInstance().isUnrestricted(e.player)) {
            sendMessage(e.player)
        }
    }

    private fun sendMessage(player: Player) {
        plugin.auth.markLoggedIn(player)
        val action = if (player.hasPlayedBefore()) MessageAction.JOIN else MessageAction.FIRST_JOIN
        plugin.messageSender.sendMessages(player, action, true)
    }

}